src/AzureOpenAIProxy.Management/Services/MetricService.cs

```diff
@@ -0,0 +1,272 @@
+using System.Data;
+using System.Data.Common;
+using AzureOpenAIProxy.Management.Database;
+using Microsoft.EntityFrameworkCore;
+using Npgsql;
+
+namespace AzureOpenAIProxy.Management.Services;
+
+public class ModelCounts
+{
+    public string? Resource { get; set; }
+    public int Count { get; set; }
+    public long PromptTokens { get; set; }
+    public long CompletionTokens { get; set; }
+    public long TotalTokens { get; set; }
+}
+
+public class MetricsData
+{
+    public string EventId { get; set; } = null!;
+    public DateTime DateStamp { get; set; }
+    public string Resource { get; set; } = null!;
+    public long PromptTokens { get; set; }
+    public long CompletionTokens { get; set; }
+    public long TotalTokens { get; set; }
+    public long Requests { get; set; }
+}
+
+public class ChartData
+{
+    public DateTime DateStamp { get; set; }
+    public long Count { get; set; }
+}
+
+public class ModelData
+{
+    public List<ModelCounts> ModelCounts { get; set; } = [];
+    public List<ChartData> ChartData { get; set; } = [];
+}
+
+public class AllEvents
+{
+    public string OrganizerName { get; set; } = null!;
+    public string EventName { get; set; } = null!;
+    public DateTime StartDate { get; set; }
+    public DateTime EndDate { get; set; }
+    public int Registered { get; set; }
+    public string EventId { get; set; } = null!;
+}
+
+
+public class MetricService() : IMetricService, IDisposable
```

Comment:
Let's be consistent and use a primary constructor and have the services injected here, rather than a default constructor.

Reply:
This problem goes away as I've moved everything to EF. The issue was conn was built from db and I didn;t want to get db twice as depending on timing they could be different.